docs/router/src/topics/outlets.md

```diff
@@ -0,0 +1,189 @@
+# Outlets
+
+[`Outlet`]s tell the router where to render content. In the following example
+the active routes content will be rendered within the [`Outlet`].
+
+```rust
+# // Hidden lines (like this one) make the documentation tests work.
+# extern crate dioxus;
+use dioxus::prelude::*;
+
+fn Index(cx: Scope) -> Element {
+    cx.render(rsx! {
+        h1 { "Index" }
+    })
+}
+
+fn App(cx: Scope) -> Element {
+    let routes = use_segment(&cx, || {
+        Segment::new().index(RcComponent(Index))
+    });
+
+    cx.render(rsx! {
+        Router {
+            routes: routes.clone(),
+            # init_only: true,
+
+            header { "header" }
+            Outlet {}
+            footer { "footer" }
+        }
+    })
+}
+#
+# let mut vdom = VirtualDom::new(App);
+# vdom.rebuild();
+# let html = dioxus::ssr::render_vdom(&vdom);
+# assert_eq!(
+#     "<header>header</header><h1>Index</h1><footer>footer</footer>",
+#     html
+# );
+```
+
+The example above will output the following HTML (line breaks added for
+readability):
+```html
+<header>
+    header
+</header>
+<h1>
+    Index
+</h1>
+<footer>
+    footer
+</footer>
+```
+
+## Nested Outlets
+When using nested routes, we need to provide equally nested [`Outlet`]s.
+
+> Learn more about [nested routes](./routes/nested.md) in their own chapter.
+
+## Named Outlets
+When building complex apps, we often need to display multiple pieces of content
+simultaneously. For example, we might have a sidebar that changes its content in
+sync with the main part of the page.
+
+When defining our routes, we can use `RcMulti` instead of `RcComponent` to tell
+the router about our content.
+
+We then can use a named [`Outlet`] in our output, to tell the router where to
+put the side content.
+
+```rust
+# // Hidden lines (like this one) make the documentation tests work.
+# extern crate dioxus;
+use dioxus::prelude::*;
+
+fn Main(cx: Scope) -> Element {
+    cx.render(rsx! {
+        main { "Main Content" }
+    })
+}
+
+fn Aside(cx: Scope) -> Element {
+    cx.render(rsx! {
+        aside { "Side Content" }
+    })
+}
+
+fn App(cx: Scope) -> Element {
+    let routes = use_segment(&cx, || {
+        Segment::new()
+            .index(RcMulti(Main, vec![("side", Aside)]))
```

Comment:
I'm starting to think I over-engineered this one. I guess current API is fine.

Reply:
I honestly have no idea if the macro variant is possible. I don't know that much about macros, pretty much only how to use them.

But I'll definitely apply your unit struct suggestion here.

Reply:
I changed it to `TypeId`s to see how it would look. I think for now it's better to stick with strings, as the other approach would further complicate route definition, and make the `Outlet` component more difficult to use.

It could be improved in a later PR though.